cs/tools/TunnelsSDK.Generator/RustContractWriter.cs

```diff
@@ -309,18 +410,19 @@ private string GetRustTypeForCSType(string csType, IPropertySymbol property, Sor
             };
         }
 
-        if (rsType.Length == 3 && rsType.StartsWith("i") || rsType.StartsWith("u"))
-        {
-            imports.Add($"std::{rsType}");
-        }
-
         if (isArray)
         {
             rsType = $"Vec<{rsType}>";
         }
 
         if (isNullable)
         {
+            if (rsType == "String")
+            {
+                imports.Add("crate::contracts::serialization::empty_string_as_none");
+                s.AppendLine("    #[serde(default, deserialize_with = \"empty_string_as_none\")]");
```

Comment:
There are certain cases where the distinction between empty string and none/null is important. For example when updating a tunnel, a null description property indicates it should not be updated, while an empty string would clear any existing description value. I can't think of any specific problems with _de_-serializing empty as null, but I'm concerned there could be some edge case in the future. Is there a benefit to doing this?

Reply:
I noticed that when creating a tunnel, the `name` and `description` come back as an empty string (rather than null/omitted) if not set.

```json
{
  "clusterId": "usw2",
  "tunnelId": "q4zmg8zl",
  "name": "",
  "description": "",
  "tags": [],
  "accessTokens": {
    "host": "eyJhbGciOiJSUzI1NiIsImtpZCI6IjNCQ0REM0ZEQTlGQzY0ODYzMjkxN0E1QjQxQzFERTJCRjIwQTU5NjUiLCJ4NXQiOiJPODNUX2FuOFpJWXlrWHBiUWNIZUtfSUtXV1UiLCJ0eXAiOiJKV1QifQ.eyJjbHVzdGVySWQiOiJ1c3cyIiwidHVubmVsSWQiOiJxNHptZzh6bCIsInNjcCI6Imhvc3QiLCJleHAiOjE2NTU1ODQwMTAsImlzcyI6Imh0dHBzOi8vdHVubmVscy5hcGkudmlzdWFsc3R1ZGlvLmNvbS8iLCJhdWQiOiJodHRwczovL3R1bm5lbHMuYXBpLnZpc3VhbHN0dWRpby5jb20vIiwianRpIjoiOGI4N2VmYjgtYzJlMi00MGJhLWE5YTYtYzQ2NDhjN2ExMGM1IiwiaWF0IjoxNjU1NDk3NjEwLCJuYmYiOjE2NTU0OTY3MTB9.VvWZe0n3YEYRURZyatTaEpICTaZOtnEYgSMFmHL-nhUvsRx_CSVlJHallyyRBdBIU3_Pyr2v2jwl0CBMwje7p5Ym3puYwtv3EXZzTxBkAoeboIQuDYxeYNFAY6gzTU9HZertiFFQVdyHskaAmwr8DT47gFprHr1-bgcBWu4rMnbjtiBGyc4kOmR04B9i7qioKRdR6NfQKN_L2KmTtK1sC_hFsfgfV5SktFsEJrr8iWbQpp181JE9y7WiIZN0vnQ0HdNyav9Og_8UCe8NAZ_wkJsCiaWrc4u6oBWg0OK7lFIWDqH4nClQZjZcRzag-0RKZKOJY8YFEx-quMS7YzHOPA"
  },
  "accessControl": { "entries": [] },
  "options": {},
  "status": {},
  "endpoints": [],
  "created": "2022-06-17T20:26:50.5466613Z"
}
```

This caused issues when we tried to generate the path for the tunnel, since we'd see there was `Some` name and try to refer to it using that name.

Reply:
Yes, a tunnel always has a description, and empty string is the default value.

An unnamed tunnel has a name value of empty string rather than null so that the update semantics can work. For example if a tunnel has a (nonempty) name, then you want to update it to remove the name, you'd send an update request with name="", because a request with name=null would indicate no change to the name.

This could be resolved another way by supporting the PATCH verb with partial JSON objects, so that updates would entirely omit properties that aren't meant to be changed. But that's more complicated for all the clients to deal with, so we haven't implemented it.

Reply:
In [JSON Merge Patch](https://datatracker.ietf.org/doc/html/rfc7386), it treated that sending `name: null` would remove the name, and omitting it from the request would leave it unchanged in an update. However not all deserializers have a way to distinguish null/omitted...

But I will remove the empty string==None logic 🙂

Reply:
> However not all deserializers have a way to distinguish null/omitted...

Right, and not all serializers have a way to omit properties. So while it could work, it adds complexity that I think wouldn't be justified at this time.